content/blog/2025-11-07-tmil-22/index.md

```diff
@@ -0,0 +1,128 @@
++++
+title = "Linebender in October 2025"
+authors = ["Raph Levien"]
++++
+Linebender is an informal open-source organization working on various projects to advance the state of the art in GUI for [the Rust programming language](https://rust-lang.org).
+
+October saw steady progress on our core crates, including a number of releases.
+
+## Vello
+
+Vello is our GPU vector renderer.
+It can draw large 2D scenes with high performance, using GPU compute shaders for most of the work.
+
+Laurenz Stampl has done his Masters degree at ETH Zurich on high performance CPU rendering of 2D graphics, particularly the techniques used in vello_cpu.
+The [masters thesis] is now published.
+It contains up-to-date background on 2D rendering, which is not well represented in the literature, as well as an explanation of the sparse strip method, and benchmarks.
+
+<figure>
+
+<img style="height: auto" width="1170" height="920" src="sparse_strip.png" alt="The tiling of a butterfly shape at four different levels of granularity. The top left shows many small tiles, while the bottom right shows fewer large ones, but also a much smaller fraction of space not covered by a tile.">
+
+<!-- <img style="height: auto; width: 50%; margin-left: 25%;" width="666" height="673" src="hybrid-blending.png" alt="A series of overlapping shapes of various colours, variously composited."> -->
+
+<figcaption>
+
+An illustration from Laurenz Stampfl's Masters Thesis, showing the effect of tile size on rendering efficiency.
+
+</figcaption>
+</figure>
+
+We released version [0.0.4 of the sparse strips][sparse strips 0.0.4] versions of the renderer.
+It should be a solid version of vello_cpu, suitable for a wider range of applications where CPU-only rendering is appropriate.
+In addition, it is the initial release of vello_hybrid, a newer approach that uses the GPU to do pixel compositing, but SIMD-accelerated geometry processing on the CPU.
+
+Thomas Smith has been making excellent progress on rendering sparse strip alpha values in GPU compute shaders.
+This was the original motivation for the sparse strip work.
+You can follow the progress in the Zulip thread [Thoughts on GPU sparse strips].
+
+Work on Vello Classic focused on support for web standard compatibility, motivated by Servo integration.
+Those features include support for non-premultiplied alpha ([vello#1173][], [vello#1262][], [vello#1145][]).
+
+Vello CPU now supports non-isolated blending ([vello#1159][] among others), which is important for HTML5 canvas compatibility.
+SVG and COLRv1 emoji, by contrast, which has set priorities for Vello in the past, only support isolated blending.
+There are also promising early results for supporting image filters including blurs.
+
+A major development in Vello Hybrid is support for multiple image atlases ([vello#1252][]), which improves performance for image rendering.
```

Comment:
I think that the order of paragraphs in this section *might* not be optimal. I bounced off it bit reading it because it jumps from Sparse Strips to Vello Classic back to sparse strips.